terraform/diff.go

```diff
@@ -373,6 +373,11 @@ type InstanceDiff struct {
 	// mean to be used for additional data a resource may want to pass through.
 	// The value here must only contain Go primitives and collections.
 	Meta map[string]interface{}
+
+	// Stub should be set when this diff exists only for purposes of providing a
+	// diff to various pre-plan or dry-run steps in the graph. A diff with this
+	// enabled should not be acted on in the plan.
+	Stub bool
```

Comment:
The way I think of it is that we shouldn't really be diffing here at all, but we're doing it as an implementation detail to make things work; ideally, we wouldn't be doing these diffs at all and maybe in future we will stop.

Therefore these diffs are not something anything outside of core should know about, and that includes the hooks. And indeed, would apply to _all_ of the hooks, to make sure that nothing outside of core ever even sees these diff objects.